src/app/core/session/session-service/remote-session.spec.ts

```diff
@@ -20,54 +29,85 @@ describe("RemoteSessionService", () => {
       },
     };
     mockHttpClient = jasmine.createSpyObj(["post", "delete"]);
+    mockHttpClient.delete.and.returnValue(of());
+
     TestBed.configureTestingModule({
       providers: [
         RemoteSession,
+        LoggingService,
         { provide: HttpClient, useValue: mockHttpClient },
       ],
     });
+
+    // Remote session allows TEST_USER and TEST_PASSWORD as valid credentials
+    dbUser = { name: TEST_USER, roles: ["user_app"] };
+    mockHttpClient.post.and.callFake((url, body) => {
+      if (body.name === TEST_USER && body.password === TEST_PASSWORD) {
+        return of(dbUser as any);
+      } else {
+        return throwError(
+          new HttpErrorResponse({ statusText: "Unauthorized" })
+        );
+      }
+    });
+
     service = TestBed.inject(RemoteSession);
   });
 
   it("should be connected after successful login", async () => {
-    expect(service.connectionState.getState()).toBe(
+    expect(service.getConnectionState().getState()).toBe(
       ConnectionState.DISCONNECTED
     );
+    expect(service.getLoginState().getState()).toBe(LoginState.LOGGED_OUT);
 
-    mockHttpClient.post.and.returnValue(of());
-
-    await service.login("", "");
+    await service.login(TEST_USER, TEST_PASSWORD);
 
     expect(mockHttpClient.post).toHaveBeenCalled();
-    expect(service.connectionState.getState()).toBe(ConnectionState.CONNECTED);
+    expect(service.getConnectionState().getState()).toBe(
+      ConnectionState.CONNECTED
+    );
+    expect(service.getLoginState().getState()).toBe(LoginState.LOGGED_IN);
   });
 
   it("should be offline if login fails", async () => {
     mockHttpClient.post.and.returnValue(throwError(new Error()));
 
-    await service.login("", "");
+    await service.login(TEST_USER, TEST_PASSWORD);
 
-    expect(service.connectionState.getState()).toBe(ConnectionState.OFFLINE);
+    expect(service.getConnectionState().getState()).toBe(
+      ConnectionState.OFFLINE
+    );
+    expect(service.getLoginState().getState()).toBe(LoginState.LOGIN_FAILED);
   });
 
   it("should be rejected if login is unauthorized", async () => {
-    const unauthorizedError = new Error();
-    unauthorizedError.name = "unauthorized";
-    mockHttpClient.post.and.returnValue(throwError(unauthorizedError));
+    await service.login(TEST_USER, "wrongPassword");
 
-    await service.login("", "");
-
-    expect(service.connectionState.getState()).toBe(ConnectionState.REJECTED);
+    expect(service.getConnectionState().getState()).toBe(
+      ConnectionState.REJECTED
```

Comment:
Could we merge ConnectionState + LoginState?

Maybe something like this could work (with some adjustments)?
- SyncedSession LoginState = LocalSession LoginState
- SyncedSession ConnectionState (if needed internally?) = RemoteSession LoginState